### PR TITLE
Add additive noise to Phi

### PR DESCRIPTION
**Pull request by jirkalhotka**
Adds additive noise from the [Unsupervised Sentence Compression using Denoising Autoencoders (CoNLL 2018), aka the New York paper](https://github.com/zphang/usc_dae) to emb2emb function. In particular, we add noise to _x_ in the Task Training below. 
image

The noise is sampled from the next batch (wrapping around to batch 0 for final batch) just like in the New York paper.

### CODE REVIEW

File: emb2emb/trainer.py [hunk 1]
```diff
@@ -257,9 +260,22 @@ def _adversarial_training(self, loss, output_embeddings, Y_embeddings):
 
         return loss, task_loss, critic_loss, train_critic_loss
 
-    def compute_emb2emb(self, Sx_batch):
+    def compute_emb2emb(self, Sx_batch, next_x_batch):
         # encode input
-        X_embeddings = self._encode(Sx_batch)
+        sent_batch =Sx_batch
```
Comment (jczestochowska) [inline emb2emb/trainer.py]: space missing

File: usc_dae_utils/operations.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import random
```
Comment (jczestochowska) [inline usc_dae_utils/operations.py:1]: I guess that's the only file we need, no?

File: usc_dae_utils/conf.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
+import argparse
```
Comment (jczestochowska) [inline usc_dae_utils/conf.py]: do we need all these files?

**Review by jczestochowska — APPROVED** [review]

File: emb2emb/trainer.py (hunk 1 above)
Comment (jirkalhotka) [reply emb2emb/trainer.py]: 2c48a68

File: usc_dae_utils/operations.py (hunk 2 above)
Comment (jirkalhotka) [reply usc_dae_utils/operations.py:1]: 2c48a68

File: usc_dae_utils/conf.py (hunk 3 above)
Comment (jirkalhotka) [reply usc_dae_utils/conf.py]: 2c48a68

**Comment by jirkalhotka** [comment]
good points @jczestochowska , it was a bit hacked from my side:) fixed in 2c48a68

**Review by jczestochowska — APPROVED** [review]

**Comment by jczestochowska** [comment]
lgtm

**Pull request merged**